src/main/java/seedu/easylog/parser/ItemsParser.java

```diff
@@ -209,4 +199,116 @@ public void processUpdateAttributeInput(String updateInput, int itemIndex, ItemM
+            ui.askForItemPriceAndStock();
+            String itemPriceAndStockInput = ui.askForUserInput();
+
+            try {
+                if (itemPriceAndStockInput == null) {
+                    throw new NullItemPriceAndStockInputException();
+                }
+
+                if (itemPriceAndStockInput.equals("")) {
+                    throw new EmptyItemPriceAndStockInputException();
+                }
+
+                String[] splitInput = itemPriceAndStockInput.split(" ");
+
+                if (splitInput.length != 2) { // magic number to be refactored later
+                    throw new IncorrectNumberOfItemPriceAndStockInputException();
+                }
+
+                String itemPriceInString = splitInput[0];
+                String itemStockInString = splitInput[1];
+
+                itemPrice = itemPriceInStringToBigDecimalFormat(itemPriceInString);
+                itemPrice = itemPrice.setScale(Constants.PRICING_SCALE, RoundingMode.HALF_EVEN);
+                itemStock = itemStockInStringToIntegerFormat(itemStockInString);
+                stopAskingUserForItemPriceAndStockInput = true;
+            } catch (NullItemPriceException e) {
+                ui.showNullItemPrice();
+            } catch (EmptyItemPriceException e) {
+                ui.showEmptyItemPrice();
+            } catch (NonNumericItemPriceException e) {
+                ui.showNonNumericItemStockInputForAdd();
+            } catch (InvalidItemPriceException e) {
+                ui.showInvalidItemPrice();
+            } catch (NullItemStockException e) {
+                ui.showNullItemStock();
+            } catch (EmptyItemStockException e) {
+                ui.showEmptyItemStock();
+            } catch (NonIntegerItemStockException e) {
+                ui.showNonIntegerItemPriceInputForAdd();
+            } catch (InvalidItemStockException e) {
+                ui.showInvalidItemStock();
+            } catch (IncorrectNumberOfItemPriceAndStockInputException e) {
+                ui.showIncorrectNumberOfItemPriceAndStockInput();
+            } catch (EmptyItemPriceAndStockInputException e) {
+                ui.showEmptyItemPriceAndStockInput();
+            } catch (NullItemPriceAndStockInputException e) {
+                ui.showNullItemPriceAndStockInput();
+            }
+        }
+
+        Item item = new Item(itemDescription, itemPrice, itemStock);
+
+        return item;
+    }
+
+    public BigDecimal itemPriceInStringToBigDecimalFormat(String itemPriceInString) throws NullItemPriceException,
+            EmptyItemPriceException, NonNumericItemPriceException, InvalidItemPriceException {
+        if (itemPriceInString == null) {
+            throw new NullItemPriceException();
+        }
+
+        String itemPriceInStringAfterTrimming = itemPriceInString.trim();
+        if (itemPriceInStringAfterTrimming.equals("")) {
+            throw new EmptyItemPriceException();
```

Comment:
same as first comment, user input is always automatically trimmed before processing the input

Reply:
I see, will make changes to it :)